src/features/broker-protection/actions/extract.js

```diff
@@ -72,7 +77,12 @@ export function extractProfiles (action, userData, root = document) {
 
     return {
         results: profilesElementList.map((element) => {
-            const scrapedData = createProfile(element, action.profile)
+            const elementFactory = (key, value) => {
+                return value?.findElements
+                    ? cleanArray(getElements(element, value.selector))
+                    : cleanArray(getElement(element, value.selector) || getElementMatches(element, value.selector))
+            }
+            const scrapedData = createProfile(elementFactory, action.profile)
```

Comment:
by moving the DOM-related operations into a function call like this, it means we can more easily test all the extraction logic without being tightly coupled to a real DOM.